### PR TITLE
SDK V2

### PR DESCRIPTION
**Pull request by cdmayer**
- Refactored to use Graph.Core library.
- Authentication removed and moved into separate repository.

This merge includes breaking changes for V1.

### CODE REVIEW

**Comment by cdmayer** [comment]
@peternied could you give this one a review?

File: src/OneDriveSdk/Requests/Extensions/IDriveSpecialCollectionRequestBuilderExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+namespace Microsoft.OneDrive.Sdk
+{
+    public partial interface IDriveSpecialCollectionRequestBuilder
+    {
+        /// <summary>
+        /// Gets app root special folder item request builder.
+        /// <returns>The item request builder.</returns>
+        /// </summary>
+        IItemRequestBuilder AppRoot { get; }
```
Comment (peternied) [inline src/OneDriveSdk/Requests/Extensions/IDriveSpecialCollectionRequestBuilderExtensions.cs]: >AppRoot [](start = 28, length = 7)

For consistency with our documentation this should be AppFolder

Comment (peternied) [inline src/OneDriveSdk/Requests/Extensions/IDriveSpecialCollectionRequestBuilderExtensions.cs]: :/ just saw this was already in use under the other type name, we should enforce the break now while we are changing everything

---
In reply to: [73946373](https://github.com/OneDrive/onedrive-sdk-csharp/pull/141#discussion_r73946373) [](ancestors = 73946373)

**Comment by peternied** [comment]
:shipit:

**Pull request merged**